Assets/Scripts/Door.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+[Serializable]
+public class Door : MonoBehaviour, IPuzzle
+{
+    [SerializeField]
+    [Tooltip("The type of door to display. 0 for wooden, 1 for barred.")]
+    [Range(0, 1)]
+    private int doorType;
+
+    [SerializeField]
+    [Tooltip("The solution to the puzzle.\nThis should be a string comprised of the connected switches' entry.")]
+    private string solution = "1";
+    
+    /// <summary>
+    /// The animator attached to the door
+    /// </summary>
+    private Animator animator;
+
+    /// <summary>
+    /// A list of all switches connected to the puzzle
+    /// </summary>
+    private List<Switch> connectedSwitches;
+
+    /// <summary>
+    /// The current attempted solution
+    /// </summary>
+    private string _currentSolution;
+
+    /// <summary>
+    /// Property to access the current solution
+    /// Compares it to the intended solution when modified
+    /// </summary>
+    private string CurrentSolution
+    {
+        get { return _currentSolution; }
+        set
+        {
+            _currentSolution = value;
+            if (DoesSolutionMatch())
+            {
+                if (CurrentSolution.Length == solution.Length)
+                    animator.SetBool("Open", true);
```

Comment:
You should practice using StringToHash to initialize some variables for your anim parameters rather than using "magic strings" and comparing strings throughout the code. https://stackoverflow.com/questions/40190241/what-is-the-benefit-of-using-animator-stringtohash